types/react/test/tsx.tsx

```diff
@@ -197,3 +197,30 @@ componentWithBadLifecycle.getSnapshotBeforeUpdate = () => { // $ExpectError
 componentWithBadLifecycle.componentDidUpdate = (prevProps: {}, prevState: {}, snapshot?: string) => { // $ExpectError
     return;
 };
+
+const Memoized1 = React.memo(function Foo(props: { foo: string }) { return null; });
+<Memoized1 foo='string'/>;
+
+const Memoized2 = React.memo(
+    function Bar(props: { bar: string }) { return null; },
+    (prevProps, nextProps) => prevProps.bar === nextProps.bar
+);
+<Memoized2 bar='string'/>;
+
+const Memoized3 = React.memo(class Test extends React.Component<{ x?: string }> {});
+<Memoized3 ref={ref => { if (ref) { ref.props.x; } }}/>;
+
+const memoized4Ref = React.createRef<HTMLDivElement>();
+const Memoized4 = React.memo(React.forwardRef((props: {}, ref: React.Ref<HTMLDivElement>) => <div ref={ref}/>));
+<Memoized4 ref={memoized4Ref}/>;
+
+const LazyClassComponent = React.lazy(async () => ({ default: ComponentWithPropsAndState }));
+const LazyMemoized3 = React.lazy(async () => ({ default: Memoized3 }));
+const LazyRefForwarding = React.lazy(async () => ({ default: Memoized4 }));
+
+<React.Suspense fallback={<Memoized1 foo='string' />}>
+    <LazyClassComponent hello='test'/>
+    <LazyClassComponent ref={ref => { if (ref) { ref.props.hello; } }} hello='test'/>
+    <LazyMemoized3 ref={ref => { if (ref) { ref.props.x; } }}/>
+    <LazyRefForwarding ref={memoized4Ref}/>
```

Comment:
😇

![image](https://user-images.githubusercontent.com/73085/47562681-47d18e80-d95a-11e8-94c6-a64722e635bb.png)

Reply:
I couldn't figure out how `ref`s work by reading `mountLazyComponent` in `ReactFiberBeginWork.js`; I actually thought they didn't work at all at first.

I did verify that they work at runtime, though (refs are set/called when the `Suspense` resolves).